### PR TITLE
Convert third_party files to utf-8 and add doc + third_party to the check

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

I noticed there were some that weren't utf-8, mostly because of the fmi ones (after adding the `/utf-8` flag to project_options on MSVC)

Conversion was done on Ubuntu (need `sudo apt install uchardet`) via:

```python
for file_path in failed_files:
    encoding = subprocess.check_output(["uchardet", file_path]).decode().strip()
    print(f"{file_path}, {encoding=}")
    ret = subprocess.check_output(['iconv', '-f', encoding, '-t', 'UTF-8', file_path, '-o', 'new'])
    print(ret)
    os.rename('new', file_path)
```

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: scripts/dev/check_non_utf8.py [hunk 1]
```diff
@@ -62,7 +62,7 @@
 # note I am skipping docs for right now; I want to do those files
```
Comment (jmarrec) [inline scripts/dev/check_non_utf8.py]:
```suggestion
```

File: scripts/dev/check_non_utf8.py [hunk 2]
```diff
@@ -59,10 +59,9 @@
 import io  # For Python 2 compat
 import sys
 
-# note I am skipping docs for right now; I want to do those files
 DIRS_TO_SKIP = [
     '.git', 'build', 'builds', 'cmake-build-debug',
-    'cmake-build-release', 'design', 'doc', 'release', 'third_party'
```
Comment (Myoldmopar) [inline scripts/dev/check_non_utf8.py:65]: Great!

**Review by Myoldmopar — APPROVED** [review]
This is awesome, and an easy win.  Including the added test coverage.  Thanks!

**Comment by Myoldmopar** [comment]
No need to wait on anything here, this is good.  Thanks @jmarrec !

**Pull request merged**